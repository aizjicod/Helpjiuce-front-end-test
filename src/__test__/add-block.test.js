import { render, screen } from '@testing-library/react';
import AddBlock from '../components/add-block';
import '../setupTests';

it('renders the right component', () => {
  const { container } = render(<AddBlock />);
  expect(container.querySelector('.add-block-title')).toBeInTheDocument();
  expect(screen.getByText(/Add blocks/)).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});
it('renders the rigth last component', () => {
  const { container } = render(<AddBlock />);
  expect(screen.getByText(/expendable Heading 1/)).toBeTruthy();
  expect(container.lastChild).toMatchSnapshot();
});
