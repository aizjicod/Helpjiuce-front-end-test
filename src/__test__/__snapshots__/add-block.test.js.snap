// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the right component 1`] = `
<div
  id="add-block"
>
  <h2
    class="add-block-title"
  >
    Add blocks
  </h2>
  <p
    class="add-block-description"
  >
    Keep typing to filter, or scape to exit
  </p>
  <ul
    class="add-block-ul"
  >
    <li
      class="container-flex"
    >
      <svg
        class="headers-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.94993 2.95002L3.94993 4.49998C3.94993 4.74851 3.74845 4.94998 3.49993 4.94998C3.2514 4.94998 3.04993 4.74851 3.04993 4.49998V2.50004C3.04993 2.45246 3.05731 2.40661 3.07099 2.36357C3.12878 2.18175 3.29897 2.05002 3.49993 2.05002H11.4999C11.6553 2.05002 11.7922 2.12872 11.8731 2.24842C11.9216 2.32024 11.9499 2.40682 11.9499 2.50002L11.9499 2.50004V4.49998C11.9499 4.74851 11.7485 4.94998 11.4999 4.94998C11.2514 4.94998 11.0499 4.74851 11.0499 4.49998V2.95002H8.04993V12.05H9.25428C9.50281 12.05 9.70428 12.2515 9.70428 12.5C9.70428 12.7486 9.50281 12.95 9.25428 12.95H5.75428C5.50575 12.95 5.30428 12.7486 5.30428 12.5C5.30428 12.2515 5.50575 12.05 5.75428 12.05H6.94993V2.95002H3.94993Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="add-block-li-information container-flex"
      >
        <p
          class="li-information-title"
        >
          Heading 1
        </p>
        <p
          class="li-information-shortcut"
        >
          Shortcut: type # + space
        </p>
      </div>
    </li>
    <li
      class="container-flex"
    >
      <svg
        class="headers-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.94993 2.95002L3.94993 4.49998C3.94993 4.74851 3.74845 4.94998 3.49993 4.94998C3.2514 4.94998 3.04993 4.74851 3.04993 4.49998V2.50004C3.04993 2.45246 3.05731 2.40661 3.07099 2.36357C3.12878 2.18175 3.29897 2.05002 3.49993 2.05002H11.4999C11.6553 2.05002 11.7922 2.12872 11.8731 2.24842C11.9216 2.32024 11.9499 2.40682 11.9499 2.50002L11.9499 2.50004V4.49998C11.9499 4.74851 11.7485 4.94998 11.4999 4.94998C11.2514 4.94998 11.0499 4.74851 11.0499 4.49998V2.95002H8.04993V12.05H9.25428C9.50281 12.05 9.70428 12.2515 9.70428 12.5C9.70428 12.7486 9.50281 12.95 9.25428 12.95H5.75428C5.50575 12.95 5.30428 12.7486 5.30428 12.5C5.30428 12.2515 5.50575 12.05 5.75428 12.05H6.94993V2.95002H3.94993Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="add-block-li-information container-flex"
      >
        <p
          class="li-information-title"
        >
          expendable Heading 1
        </p>
        <p
          class="li-information-shortcut"
        >
          Shortcut: type &gt;&gt;# + space
        </p>
      </div>
    </li>
  </ul>
</div>
`;

exports[`renders the rigth last component 1`] = `
<div
  id="add-block"
>
  <h2
    class="add-block-title"
  >
    Add blocks
  </h2>
  <p
    class="add-block-description"
  >
    Keep typing to filter, or scape to exit
  </p>
  <ul
    class="add-block-ul"
  >
    <li
      class="container-flex"
    >
      <svg
        class="headers-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.94993 2.95002L3.94993 4.49998C3.94993 4.74851 3.74845 4.94998 3.49993 4.94998C3.2514 4.94998 3.04993 4.74851 3.04993 4.49998V2.50004C3.04993 2.45246 3.05731 2.40661 3.07099 2.36357C3.12878 2.18175 3.29897 2.05002 3.49993 2.05002H11.4999C11.6553 2.05002 11.7922 2.12872 11.8731 2.24842C11.9216 2.32024 11.9499 2.40682 11.9499 2.50002L11.9499 2.50004V4.49998C11.9499 4.74851 11.7485 4.94998 11.4999 4.94998C11.2514 4.94998 11.0499 4.74851 11.0499 4.49998V2.95002H8.04993V12.05H9.25428C9.50281 12.05 9.70428 12.2515 9.70428 12.5C9.70428 12.7486 9.50281 12.95 9.25428 12.95H5.75428C5.50575 12.95 5.30428 12.7486 5.30428 12.5C5.30428 12.2515 5.50575 12.05 5.75428 12.05H6.94993V2.95002H3.94993Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="add-block-li-information container-flex"
      >
        <p
          class="li-information-title"
        >
          Heading 1
        </p>
        <p
          class="li-information-shortcut"
        >
          Shortcut: type # + space
        </p>
      </div>
    </li>
    <li
      class="container-flex"
    >
      <svg
        class="headers-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.94993 2.95002L3.94993 4.49998C3.94993 4.74851 3.74845 4.94998 3.49993 4.94998C3.2514 4.94998 3.04993 4.74851 3.04993 4.49998V2.50004C3.04993 2.45246 3.05731 2.40661 3.07099 2.36357C3.12878 2.18175 3.29897 2.05002 3.49993 2.05002H11.4999C11.6553 2.05002 11.7922 2.12872 11.8731 2.24842C11.9216 2.32024 11.9499 2.40682 11.9499 2.50002L11.9499 2.50004V4.49998C11.9499 4.74851 11.7485 4.94998 11.4999 4.94998C11.2514 4.94998 11.0499 4.74851 11.0499 4.49998V2.95002H8.04993V12.05H9.25428C9.50281 12.05 9.70428 12.2515 9.70428 12.5C9.70428 12.7486 9.50281 12.95 9.25428 12.95H5.75428C5.50575 12.95 5.30428 12.7486 5.30428 12.5C5.30428 12.2515 5.50575 12.05 5.75428 12.05H6.94993V2.95002H3.94993Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="add-block-li-information container-flex"
      >
        <p
          class="li-information-title"
        >
          expendable Heading 1
        </p>
        <p
          class="li-information-shortcut"
        >
          Shortcut: type &gt;&gt;# + space
        </p>
      </div>
    </li>
  </ul>
</div>
`;
